controls/oscal.py

```diff
@@ -54,15 +53,25 @@ class Catalog (object):
     # that singleton instance. Instead of doing `cg = Catalog(catalog_key='NIST_SP-800-53_rev4')`,
     # do `cg = Catalog.GetInstance(catalog_key='NIST_SP-800-53_rev4')`.
     @staticmethod
-    def GetInstance(catalog_key='NIST_SP-800-53_rev4'):
+    def GetInstance(catalog_key='NIST_SP-800-53_rev4', parameter_values=dict()):
         # Create a new instance of Catalog() the first time for each catalog key
         # this method is called. Keep it in memory indefinitely.
         # Clear cache only if a catalog itself changes
-        if not hasattr(Catalog, '_cached_instance_' + catalog_key):
-            setattr(Catalog, '_cached_instance_' + catalog_key, Catalog(catalog_key=catalog_key))
-        return getattr(Catalog, '_cached_instance_' + catalog_key)
+        # REMIND: we're passing in organization parameters because we need them
+        # to render control descriptions.  GovReady-Q:
+        # (1) (mostly) treats Catalogs as singletons and (2) caches rendered control
+        # guidance.
+        # So we effectively can only have one set of parameters per GovReady
+        # instance.  This will certainly need to be fixed.
+
+        catalog_instance_key = '_cached_instance_' + catalog_key
 
-    def __init__(self, catalog_key='NIST_SP-800-53_rev4'):
+        if not hasattr(Catalog, catalog_instance_key):
+            new_catalog = Catalog(catalog_key=catalog_key, parameter_values=parameter_values)
+            setattr(Catalog, catalog_instance_key, new_catalog)
+        return getattr(Catalog, catalog_instance_key)
+
+    def __init__(self, catalog_key='NIST_SP-800-53_rev4', parameter_values=dict()):
```

Comment:
`NIST_SP-800-53_rev4` could be a constant instead of the hardcoded string

Reply:
Agreed @bama4

Reply:
Made the catalog keys constants.